File: Array /121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.cpp
int maxProfit(int* prices, int pricesSize) {
    if(pricesSize==0)return 0;
    int buy=10000;
    int ans=0;
    for(int i=0;i<pricesSize;i++){
        if(buy>prices[i]){
            buy=prices[i];
        }else if(prices[i]-buy>ans){
            ans=prices[i]-buy;
        }
    }
    return ans;
}

File: Array /121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=prices[0]
        max_profit=0
        for price in prices:
            if buy>price:
                buy=price
            
            max_profit=max(max_profit,price-buy)
        
        return max_profit

        
